PyOpenWorm/simpleProperty.py

```diff
@@ -67,17 +70,61 @@ def values(self):
     def rdf(self):
         return self.conf['rdf.graph']
 
+    def identifier(self):
+        return self.link
+
     def get(self):
-        v = Variable("var" + str(id(self)))
-        self.set(v)
-        results = GraphObjectQuerier(v, self.rdf)()
-        self.unset(v)
+        results = None
+        owner = self.owner
+        if owner.defined:
+            ident = owner.identifier()
+            results = set()
+
+            if (owner.po_cache is None or owner.po_cache.cache_index !=
+                    self.conf['rdf.graph.change_counter']):
+                owner.po_cache = POCache(
+                    self.conf['rdf.graph.change_counter'], frozenset(
+                        self.rdf.predicate_objects(ident)))
+
+            for pred, obj in owner.po_cache.cache:
+                if pred == self.link:
+                    results.add(obj)
+        else:
+            v = Variable("var" + str(id(self)))
+            self._v.add(v)
```

Comment:
Could make the add and the update one method for clarity -- these things are always supposed to happen together!